pkgs/by-name/ka/kana/package.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, rustPlatform
+, meson
+, ninja
+, pkg-config
+, rustc
+, cargo
+, wrapGAppsHook4
+, desktop-file-utils
+, libadwaita
+, gst_all_1
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kana";
+  version = "1.4";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "fkinoshita";
+    repo = "Kana";
+    rev = "v${version}";
+    hash = "sha256-/Ri723ub8LMlhbPObC83bay63JuWIQpgxAT5UUYuwZI=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-Z7DpPe8/Tt8AcLjCwKbwzQTsLe6YvWBCG7DlDkkklew=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    rustPlatform.cargoSetupHook
+    rustc
+    cargo
+    wrapGAppsHook4
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    libadwaita
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-plugins-base
+    gst-plugins-bad
+    gst-plugins-good
+  ]);
+
+  meta = with lib; {
```

Comment:
```suggestion
  meta = {
```

Nested with is bad.

Reply:
I know nested with is bad, but the only thing that is nested here is the name of the maintainer, while writing "lib" everywhere is very ugly. And this is almost a convention in nixpkgs.

Reply:
> And this is almost a convention in nixpkgs.

1. You was complaining about [finalAttrs not being written convention](https://github.com/NixOS/nixpkgs/pull/292120#discussion_r1506389890), but now is complaining about `with lib;` being a consuetudinary convention? Please be consistent.
2. https://github.com/NixOS/nixpkgs/issues/208242

Reply:
>  You was complaining about [finalAttrs not being written convention or enforced guideline](https://github.com/NixOS/nixpkgs/pull/292120#discussion_r1506389890), but now you are arguing about `with lib;` being a consuetudinary convention and enforcing it? Please be consistent.

No, my request was consistent actually: I'm requesting a formal discussion about whether a particular format or feature should be adopted, and a documentation change indicating the best practice to follow. (And, if possible, automatically  suggested changes by CI.) Before that, I'm always going to follow the common practice in Nixpkgs, which I've always found clean and easy, but I'm not going to debate if that's the best practice.

I have long found that following some suggestions of newer modifications can be a problem: for example, when adding environment variables, although some committers think that we should always prefer `env.XXX`, there are also commiters suggesting me to remove the leading `env`. This concern also affects my opinion of this suggested change, because I have added more than 30 packages in the past year, but this is the first time I have received this request, and after receiving this request, my other PRs submitted at the same time still passed the review. So I doubt the progress of committers' discussions on these issues. 

Besides that, yes, the changes are good to go and applying them doesn't seem to introduce new problems for me.

Reply:
> although some committers think that we should always prefer env.XXX, there are also commiters suggesting me to remove the leading env.

Currently, attributes not recognized by (say) `stdenv.mkDerivation` are treated as environment vars to the shell.

It can bring problematic issues, like "oh, i wrote propBuildInputs instead of propagatedBuildInputs, and this bug goes undetected".

By using `env` you are explicitly saying this is an environment variable.

----------

Indeed this is one of examples given by dream2nix project to use NixOS-like modules:

https://nix-community.github.io/dream2nix/modules.html

Reply:
> Since nix.dev is NixOS-blessed documentation, their best pratices advices are authoritative

Nothing in the nix.dev link indicates that `meta = with lib;` is unacceptable - that's a separate case from `with lib;` at the top of a file.

I do not consider this worth blocking a review on.

> attributes not recognized by (say) stdenv.mkDerivation are treated as environment vars to the shell

Nit: basically [**_all_** attributes are treated as envvars](https://nixos.org/manual/nix/stable/language/derivations#optional).

> By using env you are explicitly saying this is an environment variable.

Agreed, the point of `env` is to be explicit about intention. If an attribute's purpose is to be used as an envvar and not a builder arg, then it should use `env`.

Reply:
> Nothing in the nix.dev link indicates that `meta = with lib;` is unacceptable - that's a separate case from `with lib;` at the top of a file.

`with lib;` around meta set invariably bring the issue of nested `with`. 
The most typical usage is something like 

```nix
meta = with lib; {
   maintainers = with maintainers; . . . ;
   platforms = with platforms; . . . ;
}
```

And the [nix.dev link](https://nix.dev/guides/best-practices.html#with-scopes) says, among other things:

> When more than one with used, it’s not clear anymore where the names are coming from.

Therefore, I believe that avoiding instances of `meta = with lib;` are worth removing **and** is in line with the arguments of that link.

Further, removing nested `with` and restricting `with` to single-line expressions is more manageable than removing `with` unconditionally, therefore  being a good compromise.

> I do not consider this worth blocking a review on.

Neither I.
But I will blame GitHub: it has no such a thing as a "yellow" unblocking review. 
On the other hand, I have no power to block a PR indefinitely.